### PR TITLE
Fix #145: Supports old and new forms, as well as transition

### LINKED ISSUE #145: New copyright form
New rules are in effect (we are at the beginning of a one-month transition period) during which "All Rights Reserved" is no longer part of the copyright string. The old pubrules suggests a copyright string without that phrase (but then fails to validate without it), so that people are already using it (ReSpec will stop generating it with the next release). Specberus needs to update to match.

### PR DESCRIPTION
**Pull request by astorije**

### CODE REVIEW

**Comment by astorije** [comment]
@deniak, @tripu, @darobin, who wants to be assigned to this one? :-)

File: lib/rules/headers/copyright.js [hunk 1]
```diff
@@ -9,15 +9,23 @@ exports.check = function (sr, done) {
...
-        ,   evil = "All Rights Reserved\\."
-        ,   good = "Some Rights Reserved: this document is dual-licensed, CC-BY and W3C Document License\\."
-        ,   endRex = " W3C liability, trademark,? and document use rules apply\\.$"
-        ,   rex = new RegExp(startRex + (hatesFreedom ? evil : good) + endRex)
+        ,   startRex = "^Copyright © (?:(?:199\\d|20[01]\\d)-)?" + year + " .*?W3C® \\(MIT, ERCIM, Keio, Beihang\\)"
+        ,   dualLicensed = /CC-BY/.test(text);
+
+        // See https://lists.w3.org/Archives/Public/spec-prod/2015JanMar/0001.html
+        if (sr.getDocumentDate() < new Date(2015, 01, 07)) // Before 06 Feb 2015
+            var w3cOnlyStatement = ", All Rights Reserved";
```
Comment (darobin) [inline lib/rules/headers/copyright.js]: You're declaring this variable three times (and the three declarations get hoisted to the function's top). Just declare it as part of the top block and then simply assign to it in the conditional.

File: lib/rules/headers/copyright.js [hunk 2]
```diff
@@ -9,15 +9,23 @@ exports.check = function (sr, done) {
     if ($c.length) {
         var year = (sr.getDocumentDate() || new Date).getFullYear()
         ,   text = sr.norm($c.text())
-        ,   startRex = "^Copyright © (?:(?:199\\d|20[01]\\d)-)?" + year + " .*?W3C® \\(MIT, ERCIM, Keio, Beihang\\), "
-        ,   hatesFreedom = /All Rights Reserved/.test(text)
-        ,   evil = "All Rights Reserved\\."
-        ,   good = "Some Rights Reserved: this document is dual-licensed, CC-BY and W3C Document License\\."
-        ,   endRex = " W3C liability, trademark,? and document use rules apply\\.$"
```
Comment (darobin) [inline lib/rules/headers/copyright.js]: You take all the fun out :)

**Comment by darobin** [comment]
With the minor `var` thing fixed this LGTM.

File: lib/rules/headers/copyright.js (hunk 2 above)
Comment (astorije) [inline lib/rules/headers/copyright.js]: FUN IST VERBOTEN

File: lib/rules/headers/copyright.js (hunk 1 above)
Comment (astorije) [inline lib/rules/headers/copyright.js]: Fixed in cd48bc2

**Comment by astorije** [comment]
@darobin, thanks! I wasn't sure regarding the dates to compare to (aka will people be upset when their document fails on March 8...).

**Comment by darobin** [comment]
I guess we'll see when this happens. It has been announced and there's been a transition period. The tools are updated (at least ReSpec is, @tabatkins can confirm for Bikeshed). We should be okay!

**Comment by mikewest** [comment]
Bikeshed is updated as of https://github.com/tabatkins/bikeshed/commit/d80da506293c08b3effb85cfa950be8b7332ade1.

**Comment by darobin** [comment]
Thanks @mikewest! Well, @astorije, nothing stands in the way of merging :)

**Pull request merged**